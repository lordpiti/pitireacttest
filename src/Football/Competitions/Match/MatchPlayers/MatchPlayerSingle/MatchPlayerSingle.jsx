import React from 'react';

const MatchPlayerSingle = ( props ) => {

  let substitutionIn, substitutionOut = null;

  if (props.player.substitutionIn) {
    substitutionIn = <span>
      <span className="glyphicon glyphicon-arrow-left"></span>
      <span>{props.player.substitutionIn.minute}</span>
    </span>;
  }

  if (props.player.substitutionOut) {
    substitutionOut = <span>
      <span className="glyphicon glyphicon-arrow-right"></span>
      <span>{props.player.substitutionOut.minute}</span>
    </span>;
  }

  return (
    <div className="row match-player-single">
    <div className="col-sm-1">{props.player.dorsal}</div>
    <div className="col-sm-5">
      {props.player.name} {props.player.surname}
    </div>
    <div className="col-sm-1">
        {substitutionIn}
        {substitutionOut}
    </div>
    <div className="col-sm-2">
      {props.player.bookings.map((booking, index) => {
        const cardImageUrl = booking.type==='Amarilla'?'/assets/img/yellow_card-512.png':'/assets/img/red_card-512.png';

        return <div key={index}>
          <img src={cardImageUrl} width="20" height="20" alt=""/>
          <label className="small">{booking.minute}</label>
        </div>
      }
      )}
    </div>
    <div className="col-sm-2">
      {props.player.goals.map((goal, index) =>
        <span key={index}>
          <img src="/assets/img/ball.png" width="20" height="20" alt=""/>
          <label className="small">{goal.minute}</label>
        </span>)
      }
    </div>
  </div>
  )
};

export default MatchPlayerSingle;