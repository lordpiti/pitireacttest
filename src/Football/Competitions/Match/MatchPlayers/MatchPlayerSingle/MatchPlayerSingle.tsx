import React from 'react';

interface PlayerMatchSingle {
  name: string;
  surname: string;
  dorsal: string;
  substitutionIn: Substitution;
  substitutionOut: Substitution;
  bookings: Booking[];
  goals: Goal[];
}

interface Booking {
  minute: number;
  type: string;
}

interface Goal {
  minute: number;
}

interface MatchPlayerSingleProps {
  player: PlayerMatchSingle;
}

interface Substitution {
  minute: number;
}

const MatchPlayerSingle = (props: MatchPlayerSingleProps) => {
  return (
    <div className='row match-player-single'>
      <div className='col-sm-1'>{props.player.dorsal}</div>
      <div className='col-sm-5'>
        {props.player.name} {props.player.surname}
      </div>
      <div className='col-sm-1'>
        {props.player.substitutionIn && (
          <span>
            <span className='glyphicon glyphicon-arrow-left'></span>
            <span>{props.player.substitutionIn.minute}</span>
          </span>
        )}
        {props.player.substitutionOut && (
          <span>
            <span className='glyphicon glyphicon-arrow-right'></span>
            <span>{props.player.substitutionOut.minute}</span>
          </span>
        )}
      </div>
      <div className='col-sm-2'>
        {props.player.bookings.map((booking, index) => {
          const cardImageUrl =
            booking.type === 'Amarilla'
              ? '/assets/img/yellow_card-512.png'
              : '/assets/img/red_card-512.png';

          return (
            <div key={index}>
              <img src={cardImageUrl} width='20' height='20' alt='' />
              <label className='small'>{booking.minute}</label>
            </div>
          );
        })}
      </div>
      <div className='col-sm-2'>
        {props.player.goals.map((goal, index) => (
          <span key={index}>
            <img src='/assets/img/ball.png' width='20' height='20' alt='' />
            <label className='small'>{goal.minute}</label>
          </span>
        ))}
      </div>
    </div>
  );
};

export default MatchPlayerSingle;
