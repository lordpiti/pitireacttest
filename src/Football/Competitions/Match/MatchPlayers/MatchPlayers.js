import React from 'react';

const MatchPlayers = ( props ) => {
  return (
    <div>
      <div>
      {props.matchPlayers.filter(player => player.teamId === props.team.id && player.start).map(player =>
        <div>{player.name}</div>
      )}
      </div>
      <hr />
      <div>
        {props.matchPlayers.filter(player => player.teamId === props.team.id && !player.start).map(player =>
          <div>{player.name}</div>
        )}
      </div>
    </div>
  )
};

export default MatchPlayers;