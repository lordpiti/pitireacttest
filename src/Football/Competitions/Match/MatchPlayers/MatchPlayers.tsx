import React from 'react';
import MatchPlayerSingle from './MatchPlayerSingle/MatchPlayerSingle';

interface MatchPlayersProps {
  team: any;
  matchPlayers: any[];
}

const MatchPlayers = (props: MatchPlayersProps) => {
  return (
    <div>
      <div>
        {props.matchPlayers
          .filter((player) => player.teamId === props.team.id && player.start)
          .map((player) => (
            <MatchPlayerSingle key={player.playerId} player={player} />
          ))}
      </div>
      <hr />
      <div>
        {props.matchPlayers
          .filter((player) => player.teamId === props.team.id && !player.start)
          .map((player) => (
            <MatchPlayerSingle key={player.playerId} player={player} />
          ))}
      </div>
    </div>
  );
};

export default MatchPlayers;
