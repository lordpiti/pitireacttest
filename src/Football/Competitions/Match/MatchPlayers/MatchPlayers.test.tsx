import React from 'react';
import { shallow } from 'enzyme';
import MatchPlayers from './MatchPlayers';
import MatchPlayerSingle from './MatchPlayerSingle/MatchPlayerSingle';

describe('<MatchPlayers />', () => {
  it('should render two MatchPlayerSingle items', () => {
    const playerList = [
      { playerId: 1, teamId: 1, start: true },
      { playerId: 2, teamId: 1, start: false },
    ];
    const team = { id: 1 };
    const wrapper = shallow(
      <MatchPlayers team={team} matchPlayers={playerList} />
    );
    expect(wrapper.find(MatchPlayerSingle)).toHaveLength(2);
  });
});

describe('renders the snapshot', () => {
  it('should render one MatchPlayerSingle', () => {
    const playerList = [
      { playerId: 1, teamId: 1, start: true },
      { playerId: 2, teamId: 1, start: false },
    ];
    const team = { id: 1 };

    const tree = shallow(
      <MatchPlayers team={team} matchPlayers={playerList} />
    );

    expect(tree).toMatchSnapshot();
  });
});
