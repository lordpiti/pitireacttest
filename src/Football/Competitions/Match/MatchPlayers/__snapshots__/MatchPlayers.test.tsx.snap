// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the snapshot should render one MatchPlayerSingle 1`] = `
<div>
  <div>
    <MatchPlayerSingle
      key="1"
      player={
        Object {
          "playerId": 1,
          "start": true,
          "teamId": 1,
        }
      }
    />
  </div>
  <hr />
  <div>
    <MatchPlayerSingle
      key="2"
      player={
        Object {
          "playerId": 2,
          "start": false,
          "teamId": 1,
        }
      }
    />
  </div>
</div>
`;
